platform-sdk/docs/core/dynamic-address-book/testing/GreenWire-Test-Plan.md

```diff
@@ -0,0 +1,116 @@
+        * when changing an event's birth round the hash of the event will not be modified.
+
+# New Edge Cases
+
+### Minimum Birth Round of Judges
+
+Initially the `AncientThreshold` was implemented as `pendingConsensusRound - roundsNonAncient` but an edge case is
+possible where more than `roundsNonAncient` number of rounds come to consensus all at once causing critical events to
+become ancient too soon. The definition of ancient for birth rounds was taken as the minimum birth round of the judges
+to fix this.
+
+### Initial EventWindow
+
+The initial `AncientThreshold` in the two modes is different. In `GENERATION_THRESHOLD` mode, the
+initial `AncientThreshold` is `EventConstants.FIRST_GENERATION = 0` and in `BIRTH_ROUND_THRESHOLD` mode, the
+initial `AncientThreshold` is `ConsensusConstants.ROUND_FIRST = 1`.
+
+### Sequence Data Structures
+
+The sequential data structures need to be initialized with different method accessors for the two modes.
+In `GENERATION_THRESHOLD` mode, the data structures are initialized with `Event :: getGeneration()` and
+in `BIRTH_ROUND_THRESHOLD` mode, the data structures are initialized with `Event :: getBirthRound()`
+
+# Component Unit Tests
+
+These unit tests are implemented in the PR that modifies the component to handle birth round as ancient.
+
+Component unit tests are expanded to handle both modes, either through parameterized tests or by creating separate unit
+tests for each mode.
+The Component's unit tests cover the following cases:
+
+* The ability of the component to receive events with birth rounds in addition to generation
+* The ability of the component to receive EventWindows containing the AncientThreshold.
+* The ability of the component to behave appropriately when an event is determined to be ancient by the event window.
+    * The behavior of the component must be the same in both modes, i.e. that it handles ancient events in the same way.
+
+# Intake Unit Tests
+
+Intake unit tests wire multiple components together to test the pipeline for receiving events from gossip all the way
+through to consensus.
+
+## TestFixture Modifications
+
+### GraphGenerator
+
+The `GraphGenerator` is modified to have an internal instance of `ConsensusImpl` so it can give events their accurate
+birth round.
+
+* Before each event is created, the `pendingConsensusRound` is queried from the internal `ConsensusImpl`
+* The created event receives a birth round equal to `pendingConsensusRound`
+* The event is added to the internal `ConsensusImpl` which may cause the `pendingConsensusRound` to increment.
+* The event is then emitted for use in testing.
+
+### TestIntake
+
+The `TestIntake` class is modified to receive the `AncientMode` as a constructor argument and sets up the internal
+pipeline of components in the same ancient mode.
+
+### ConsensusTests
+
+The `ConsensusTests` are modified to receive an additional parameter for the `AncientMode`.
```

Comment:
This definitely needs more detail

Reply:
I've tabled this work until turtle lite is available and will detail testing at that time.